examples/silentpayments.c

```diff
@@ -0,0 +1,503 @@
+         */
+        sp_addresses[0] = &carol_address; /* : 1.0 BTC */
+        sp_addresses[1] = &bob_address;   /* : 2.0 BTC */
+        sp_addresses[2] = &carol_address; /* : 3.0 BTC */
+        for (i = 0; i < N_TX_OUTPUTS; i++) { ret =
+            secp256k1_ec_pubkey_parse(ctx,
+                &recipients[i].scan_pubkey,
+                (*(sp_addresses[i]))[0],
+                33
+            );
+            assert(ret);
+            ret = secp256k1_ec_pubkey_parse(ctx,
+                &recipients[i].spend_pubkey,
+                (*(sp_addresses[i]))[1],
+                33
+            );
+            assert(ret);
+            /* Alice creates the recipient objects and adds the index of the
+             * original ordering (the ordering of the `sp_addresses` array) to
+             * each object. This index is used to return the generated outputs
+             * in the original ordering so that Alice can match up the generated
+             * outputs with the correct amounts.
+             */
+            recipients[i].index = i;
+            recipient_ptrs[i] = &recipients[i];
+        }
+        for (i = 0; i < N_TX_OUTPUTS; i++) {
+            generated_output_ptrs[i] = &generated_outputs[i];
+        }
+        ret = secp256k1_silentpayments_sender_create_outputs(ctx,
+            generated_output_ptrs,
+            recipient_ptrs, N_TX_OUTPUTS,
+            smallest_outpoint,
+            sender_seckey_ptrs, N_TX_INPUTS,
+            NULL, 0
+        );
+        assert(ret);
+        printf("Alice created the following outputs for Bob and Carol: \n");
+        for (i = 0; i < N_TX_OUTPUTS; i++) {
+            printf("    ");
+            printf("%s : ", address_amounts[i]);
+            secp256k1_xonly_pubkey_serialize(ctx,
+                xonly_print,
+                &generated_outputs[i]
+            );
+            print_hex(xonly_print, sizeof(xonly_print));
+        }
+        /* Store the generated outputs in the `tx_outputs` array. The
+         * `tx_outputs` array is used to represent the final transaction, which
+         * is what Bob and Carol would use for scanning.
+         */
+        for (i = 0; i < N_TX_OUTPUTS; i++) {
+            tx_outputs[i] = generated_outputs[i];
+        }
+
+        /* It's best practice to try to clear secrets from memory after using
+         * them. This is done because some bugs can allow an attacker to leak
+         * memory, for example through "out of bounds" array access (see
+         * Heartbleed), or the OS swapping them to disk. Hence, we overwrite the
+         * secret key buffer with zeros.
+         *
+         * Here we are preventing these writes from being optimized out, as any
+         * good compiler will remove any writes that aren't used. */
+        secure_erase(seckey, sizeof(seckey));
+        for (i = 0; i < N_TX_INPUTS; i++) {
+            secure_erase(&sender_seckeys[i], sizeof(sender_seckeys[i]));
+        }
+    }
+
```

Comment:
The example file can also show how to use the `secp256k1_silentpayments_recipient_create_labelled_spend_pubkey` function.
Here is a suggestion.

```suggestion
{
        /*** Label ***/

        unsigned char label_tweak[32];
        secp256k1_pubkey label;
        secp256k1_pubkey labelled_spend_pubkey;
        secp256k1_pubkey bob_spend_ec_pubkey;
        unsigned char labelled_spend_pubkey_print[33];
        size_t outputlen = 33;

        ret = secp256k1_ec_pubkey_parse(ctx, &bob_spend_ec_pubkey, bob_spend_pubkey, sizeof(bob_spend_pubkey));
        assert(ret);

        ret = secp256k1_silentpayments_recipient_create_label_tweak(ctx, &label, label_tweak, bob_scan_key, 1);
        assert(ret);
        ret = secp256k1_silentpayments_recipient_create_labelled_spend_pubkey(ctx, &labelled_spend_pubkey, &bob_spend_ec_pubkey, &label);
        assert(ret);

        ret = secp256k1_ec_pubkey_serialize(ctx, labelled_spend_pubkey_print, &outputlen, &labelled_spend_pubkey, SECP256K1_EC_COMPRESSED);
        assert(ret);
        printf("Labelled Bob Spend Pubkey: ");
        print_hex(labelled_spend_pubkey_print, sizeof(labelled_spend_pubkey_print));
        assert(memcmp(labelled_spend_pubkey_print, bob_address[1], 33) == 0);
        printf("It is the same as bob_address[1]. This is how we derive labeled addresses.\n");
    }
    
```

Reply:
Thanks for the suggestion! I tried a few ways to work this into the example but was never satisfied with the result. It always ended up making the example more complicated (first create bob's address, then do the sending, then do the receiving), and I don't think it adds much since this function is fairly self explanatory.